purpleair/channel.py

```diff
@@ -149,13 +157,24 @@ def setup(self) -> None:
         self.created: Optional[int] = self.channel_data.get('Created')
         self.uptime: Optional[int] = self.channel_data.get('Uptime')
         self.is_owner: Optional[bool] = bool(self.channel_data.get('isOwner'))
-
+    
+    def get_all_historical(self,
+                           weeks_to_get: int,
+                           start_date: datetime = datetime.now(),
+                           thingspeak_args={}) -> pd.DataFrame:
```

Comment:
Do not use a `{}` as a default; each invocation reuses the same `{}`:

```
>>> def test(k: int, v: str, d = {}):
...     d[k] = v
...     return d
...
>>> test(1, 'a')
{1: 'a'}
>>> test(2, 'b')
{1: 'a', 2: 'b'}
```

Reply:
Removed passing dict